packages/block-editor/src/components/block-variation-transforms/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { SelectControl } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+function __experimentalBlockVariationTransforms( {
+	blockName,
+	selectedBlockClientId,
+} ) {
+	const [ variationsSelectValue, setVariationsSelectValue ] = useState( '' );
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+	const { variations } = useSelect(
+		( select ) => {
+			const { getBlockVariations } = select( 'core/blocks' );
+			const scope = 'transform';
+			const _variations = getBlockVariations(
+				blockName,
+				scope
+			).filter( ( variation ) => variation.scope?.includes( scope ) );
+			return {
+				variations: _variations,
+			};
+		},
+		[ blockName ]
+	);
+	if ( ! variations.length ) return null;
+
+	const selectOptions = [
+		{ value: '', label: __( 'Select a variation' ) },
+		...variations.map( ( variation ) => ( {
+			value: variation.name,
+			label: variation.title || 'no title??? change me',
+		} ) ),
+	];
+	const onSelectVariation = ( variationName ) => {
+		setVariationsSelectValue( variationName );
+		if ( ! variationName ) return;
```

Comment:
How is it possible? Shouldn't be the option that acts like a label disabled so users couldn't pick them?

Reply:
This is just temporary as the design will change and probably will not be a simple select.